### DISCUSSION TITLE
Is it possible to write queries for a CSV file?

### DISCUSSION THREAD

**Discussion by Holden121** [dbeaver/dbeaver#11189] [Q&A] [answered]
Hello all,

Is it possible to write queries for a CSV file?

Here is the long story:

I am new to DBeaver. I discovered that I can use a CSV file like a table. So I got one, and I am able to see it in DBeaver.

I then tried to write a simple query to create a variable [omegle](https://omegle.onl/) [xender](https://xender.vip/) for joining CSV data with a table on an SQL server.

On CSV I have the last names, first names, and birth dates (DOB) of individuals. Same on the table on the SQL server. In cases like this, I usually create a variable made up of the concatenation of the first four letters of the first and last names, plus the DOBS.

This is where the problem started: The function LEFT does not seem to work on the first and last names.

Any comments would be appreciated.

Thanks,

**Comment by tati-kru** [comment, marked as answer]
DBeaver works with CSV through the CSV JDBC driver. This driver supports very basic SQL dialect that does not include the LEFT function. You need to check what functions is available for this driver or use any other CSV driver. As we know some commercial CSV drivers suggests richer functionality.
